src/scripts/content.js

```diff
@@ -18,17 +18,57 @@ const createTeamsNotificationScript = () => {
   (document.head || document.documentElement).appendChild(s);
 };
 
-chrome.storage.sync.get(defaultOptions, optionsFromStorageOrDefaults => {
-  teamsExtension.options = optionsFromStorageOrDefaults;
-  console.debug('Teams extension options', teamsExtension.options);
-
-  createTeamsNotificationScript();
-});
+const createFocusLostHandler = windowId => () => {
+  console.debug('Focus lost: ' + windowId);
+  if (teamsExtension.options.switchToNewChatOnBlur) {
+    // This timeout is necessary to be able to focus a Teams window by clicking its notification. Otherwise it just minimizes right away.
+    setTimeout(() => {
```

Comment:
It is not clear to me what is the purpose of this timeout. Can you explain me better? What is the notification mentioned?

Reply:
to be honest, I just copied this piece of code from @janhalasa fork and it's branch swtich-to-activity-tab-on-blur